### PR TITLE
Support OneOf for parameter validation

### PR DESCRIPTION
**Pull request by vladbarosan**
- Support oneOf for parameter validation

### CODE REVIEW

**Comment by amarzavery** [comment]
looks like some linting error.

File: lib/types/parameter-value.js [hunk 1]
```diff
@@ -162,9 +162,26 @@ function ParameterValue (parameterObject, raw) {
...
             // Convert/Coerce the raw value from the request object
             try {
-              processedValue = helpers.convertValue(schema, {
-                collectionFormat: parameterObject.collectionFormat
-              }, raw);
+              if (schema.oneOf) {
+                for (var oneOfSchema of schema.oneOf) {
+
+                  // There is no type "null".
+                  if (oneOfSchema.type == "null") {
```
Comment (amarzavery) [inline lib/types/parameter-value.js]: ===

**Comment by vladbarosan** [comment]
yes its because the linter doesnt accept es6, ill configure it to do so

File: lib/types/parameter-value.js (hunk 1 above)
Comment (amarzavery) [reply lib/types/parameter-value.js]: I think the linter is failing for not having `===`. If you do this and run the linter locally then you should be good I guess.

File: lib/types/parameter-value.js [hunk 2]
```diff
@@ -162,9 +162,26 @@ function ParameterValue (parameterObject, raw) {
           } else {
             // Convert/Coerce the raw value from the request object
             try {
-              processedValue = helpers.convertValue(schema, {
-                collectionFormat: parameterObject.collectionFormat
-              }, raw);
+              if (schema.oneOf) {
+                for (var oneOfSchema of schema.oneOf) {
```
Comment (amarzavery) [inline lib/types/parameter-value.js]: Oh and it could be because of `for ... of` as well.

**Pull request merged**